selfdrive/camerad/cameras/camera_qcom.cc

```diff
@@ -2055,96 +2045,111 @@ static void* ops_thread(void* arg) {
   return NULL;
 }
 
-void camera_process_front(MultiCameraState *s, CameraState *c, int cnt) {
-  common_camera_process_front(s->sm_front, s->pm, c, cnt);
-}
-
-// called by processing_thread
-void camera_process_frame(MultiCameraState *s, CameraState *c, int cnt) {
-  const CameraBuf *b = &c->buf;
+static void update_lapmap(MultiCameraState *s, const CameraBuf *b, const int cnt) {
+  const size_t width = b->rgb_width / NUM_SEGMENTS_X;
+  const size_t height = b->rgb_height / NUM_SEGMENTS_Y;
+  static std::unique_ptr<uint8_t[]> rgb_roi_buf = std::make_unique<uint8_t[]>(width * height * 3);
+  static std::unique_ptr<int16_t[]> conv_result = std::make_unique<int16_t[]>(width * height);
 
   // sharpness scores
   const int roi_id = cnt % ARRAYSIZE(s->lapres);  // rolling roi
-  int roi_x_offset = roi_id % (ROI_X_MAX-ROI_X_MIN+1);
-  int roi_y_offset = roi_id / (ROI_X_MAX-ROI_X_MIN+1);
```

Comment:
It looks like this is a previous bug.it should be` roi_id / (ROI_X_MAX-ROI_X_MIN+1);`

Reply:
it should be correct. col(x) is the mod and row(y) is the integer division of stride, no?

Reply:
my bad.reverted